### DISCUSSION TITLE
Any way to avoid frames going to RAM cache and just to Stream videos with GPU?

### DISCUSSION THREAD

**Discussion by AshhadQazi** [bonsai-rx/bonsai#1960] [Q&A]
Hello!

Currently we are using 3 of FLIR's Blackflys, all recording at 500 fps. While using the normal video writer for all 3, we are only really able to get around 5 minutes of footage before our RAM fills up, causing a loss of frames. The RAM is really the only issue, and is bottlenecking the entire process. The GPU and CPU barely get used throughout recording, is there any way to change Bonsai to use more of the GPU via streaming?

**Comment by bruno-f-cruz** [comment]
Heya!

This topic has been discussed several times. 

Assuming you are using the OpenCV operator, the bottleneck is either in CPU (Are you sure that a couple of cores are not getting 100%'ed?) or writing bandwidth (are you using an HD instead SSD?). I have never seen a case where RAM is the bottleneck for processing.

Assuming, CPU is the issue (usually is), you can use a GPU encoding approach using ffmpeg. Look for other posts and continue the discussion there (e.g. https://github.com/orgs/bonsai-rx/discussions?discussions_q=is%3Aopen+ffmpeg). I am closing this one for now to start preventing a lot of duplication and centralize discussion. Thanks
